File: DietiDeals24Server/src/main/java/com/dietideals24/DietiDeals24/entity/Offer.java
package com.dietideals24.DietiDeals24.entity;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "offer")
public class Offer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long offerId;
    @Column (nullable = false)
    private long bidderId;
    @Column (nullable = false)
    private double bidAmount;
    @Column (nullable = false)
    private long auctionId;
}

File: DietiDeals24/DietiDeals24/Model/Offer.swift
//
//  Offer.swift
//  DietiDeals24
//
//  Created by Antonio Abbatiello on 29/03/24.
//

import Foundation

struct Offer: Codable {
    let offerId: Int?
    var bidderId: Int
    var bidAmount: Double
    var auctionId: Int
    
    init(offerId: Int?, bidderId: Int, bidAmount: Double, auctionId: Int) {
        self.offerId = offerId
        self.bidderId = bidderId
        self.bidAmount = bidAmount
        self.auctionId = auctionId
    }
}
